packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -34,9 +34,28 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
   const cookie = ref<T | undefined>(cookies[name] as any ?? opts.default?.())
 
   if (process.client) {
-    const callback = () => { writeClientCookie(name, cookie.value, opts as CookieSerializeOptions) }
+    const channel = new BroadcastChannel(`nuxt:cookies:${name}`)
```

Comment:
However, there is [small percentage](https://caniuse.com/broadcastchannel) of browsers not supporting this API, i would add a check guard  if feature is not available.